### DISCUSSION TITLE
Integrate with GitHub

### DISCUSSION THREAD

**Discussion by gr2m** [excalidraw/excalidraw#2716] [General]
I used Excalidraw for the first time and love it! I used it to create a diagram for @probot: 
https://github.com/probot/probot/pull/1453

I've seen these related issues: https://github.com/excalidraw/excalidraw/issues/2611, https://github.com/excalidraw/excalidraw/issues/2060, but I don't only want to suggest a feature but also offer to help implement / maintain it. I'm a maintainer of @octokit and @probot, I know the GitHub platform APIs very well and am building all the JS tooling myself.

In a nutshell, I'd like to simplify the process to collaborate on Excalidraw diagrams that are part of Open Source project hosted on GitHub. 

Currently a user would need to 

1. Download the source `*.json` file
2. Upload the `*.json` file to https://excalidraw.com/
3. Make edits
4. Download `*.json` file. Export `.png`/`.svg` file as static output for comparison
5. Fork the repository
6. Update the `.json` and `.png`/`.svg` file for the diagram
7. Send a pull request
8. On every feedback loop, repeat step `1.` - `4.` and `6.`

Ideally, I'd like an <kbd>edit on Excalidraw</kbd> button, which would open https://excalidraw.com with the `.json` file already loaded. The Excalidraw UI could be altered slightly to include a button to include a <kbd>update on GitHub</kbd> button which would either create or update the pull request.

The created pull request would include a  <kbd>edit on Excalidraw</kbd> button which would load the current PR's version of the diagram.

I'm passionate about enabling/simplifying new kinds of contributions to Open Source projects, so happy to help with this one, if you are interested. 

As a side note, if you implement a GitHub App, you can put it on the marketplace and set up paid plans, e.g. if folks want to use it for private repositories. I don't expect it to be a huge source of revenue, but you could funnel the money into your existing Open Collective. It wouldn't be too much work to set this up once you have a GitHub App anyway.

**Comment by dwelle** [comment]
Sounds great. Where do we start? :smile: 

- For the <kbd>open on Excalidraw</kbd> — you give us an `text/plain` or `application/json` url-accessible JSON file and we're good to go.
- For the <kbd>update on GitHub</kbd>~~, how's that gonna work? We communicate directly with GH? I guess that would require users supplying their access tokens? Or do you have some other workflow in mind?~~ nvm, I forgot we're talking about OSS and PRs here :)

**Reply by gr2m** [reply]
I think the "open in Excalidraw" feature would be a good place to start, and would be useful beyond an integration with GitHub. I looked through open issues but haven't found any about a request to support opening Excalidraw with a URL query parameter which would load the external JSON file into the editor. I'd say we start with that? Is there any previous discussion / work on such a feature?

If not, could you or one of the maintainers create one with a few pointers on how you would get started to work on it? I'd be happy to try implementing it, or to support anyone who would like to do it themselves

**Reply by dwelle** [reply]
https://github.com/excalidraw/excalidraw/pull/2726

**Comment by wong2** [comment]
Can I help with this?

**Reply by dwelle** [reply]
@wong2 missed your comment. If you still want to work on this, lemme know.
